src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java

```diff
@@ -29,83 +30,73 @@
-            final FreeCamModule freeCam = (FreeCamModule) Seppuku.INSTANCE.getModuleManager().find(FreeCamModule.class);
-
-            if(freeCam != null && freeCam.isEnabled()) {
-                return;
-            }
+        final boolean instant = placeDelay.getValue() == 0;
+        if (this.placeTimer.passed(this.placeDelay.getValue()) || instant) {
+            if (event.getStage() == EventStageable.EventStage.PRE) {
+                final FreeCamModule freeCam = (FreeCamModule) Seppuku.INSTANCE.getModuleManager().find(FreeCamModule.class);
 
-            final Vec3d pos = MathUtil.interpolateEntity(mc.player, mc.getRenderPartialTicks());
-            final float playerSpeed = (float) MathUtil.getDistance(pos, mc.player.posX, mc.player.posY, mc.player.posZ);
+                if (freeCam != null && freeCam.isEnabled()) {
+                    return;
+                }
 
-            final BlockPos interpPos = new BlockPos(pos.x, pos.y, pos.z);
+                final Vec3d pos = MathUtil.interpolateEntity(mc.player, mc.getRenderPartialTicks());
+                final float playerSpeed = (float) MathUtil.getDistance(pos, mc.player.posX, mc.player.posY, mc.player.posZ);
 
-            final BlockPos north = interpPos.north();
-            final BlockPos south = interpPos.south();
-            final BlockPos east = interpPos.east();
-            final BlockPos west = interpPos.west();
+                final BlockPos interpPos = new BlockPos(pos.x, pos.y, pos.z);
 
-            final BlockPos northBelow = interpPos.north().down();
-            final BlockPos southBelow = interpPos.south().down();
-            final BlockPos eastBelow = interpPos.east().down();
-            final BlockPos westBelow = interpPos.west().down();
+                final BlockPos north = interpPos.north();
+                final BlockPos south = interpPos.south();
+                final BlockPos east = interpPos.east();
+                final BlockPos west = interpPos.west();
 
-            int lastSlot;
-            final int slot = findStackHotbar(Blocks.OBSIDIAN);
-            if (hasStack(Blocks.OBSIDIAN) || slot != -1) {
-                if ((mc.player.onGround && playerSpeed <= 0.005f)
-                        && (this.sneak.getValue() || (!mc.gameSettings.keyBindSneak.isKeyDown()))) {
+                final BlockPos[] surroundBlocks = {north.down(), south.down(), east.down(), west.down(),
+                        north, south, east, west};
 
-                    lastSlot = mc.player.inventory.currentItem;
-                    mc.player.inventory.currentItem = slot;
-                    mc.playerController.updateController();
+                int lastSlot = 0;
+                final int slot = findStackHotbar(Blocks.OBSIDIAN);
+                if (hasStack(Blocks.OBSIDIAN) || slot != -1) {
+                    if ((mc.player.onGround && playerSpeed <= 0.005f) && (this.sneak.getValue() || (!mc.gameSettings.keyBindSneak.isKeyDown()))) {
```

Comment:
What's with the 0.005?  Magic numbers add more confusion to code without a clear description of what it's for.

Reply:
This code was here before I ever got it, I just formatted it, so I'm not exactly sure.

Reply:
Where there's a chance to improve code, you should always take it.  You can try to come up with a better way to use a speed here, or just check if we are standing still.

Reply:
I experimented for a bit, and from what I've seen this probably is the best method.
Setting the threshold to `0f` will result in no blocks placing.
I also tried `player.velocityChanged`, but results aren't consistent since it only factors velocity.
There's no `player.positionChanged`, and writing a whole function for this one specific purpose would likely be less than ideal.

I guess what this is comments are for.

Reply:
I made this. It is meant to check if the player is going slow enough to place the blocks (If the player is going below 0.005 blocks per tick, it will place). I found it is a good balance between placing too early and not placing early enough. We could just check if the player isn't moving, but I found that didn't feel right when using the module, as it takes time for the player to slow to a stop.
I tried making this an option, but when I changed it in game, it didn't seem to change much if anything with how the module feels.